src/flow-manager.c

```diff
@@ -667,14 +667,13 @@ static TmEcode FlowManagerThreadInit(ThreadVars *t, const void *initdata, void *
     /* set the min and max value used for hash row walking
      * each thread has it's own section of the flow hash */
     uint32_t range = flow_config.hash_size / flowmgr_number;
-    if (ftd->instance == 0)
-        ftd->max = range;
-    else if ((ftd->instance + 1) == flowmgr_number) {
-        ftd->min = (range * ftd->instance) + 1;
+
+    ftd->min = ftd->instance * range;
+    ftd->max = (ftd->instance + 1) * range;
+
+    /* last flow-manager takes on hash_size % flowmgr_number extra rows */
+    if ((ftd->instance + 1) == flowmgr_number) {
         ftd->max = flow_config.hash_size;
```

Comment:
Thank you @awelzel for your first contribution to our project! :)
Just adding my thoughts around rounding of `range` that Victor asked to check.

```
hash size = 9000
thread count = 57
range (uint32_t) = 9000/57 = 157.8... but stored as 157
But, range x thread_count (our max) = 157 x 57 = 8949
So, we're short of 9000 - 8949 = 51
```
Although, with the patch, while we're at the last stage when thread count == 57, we'll set the max to 9000 so seems all will be OK..

Reply:
I think if rounding errors can get us into a state of missing rows, or assigning duplicate rows to threads, we need a new approach.

Reply:
> Thank you @awelzel for your first contribution to our project! :)
Thanks!

> Although, with the patch, while we're at the last stage when thread count == 57, we'll set the max to 9000 so seems all will be OK..

Yes, so that was the intention. The last flow-manager just picks up all remaining rows. As in the comment, it'll deal with `hash_size % flowmgr_number` more rows than the others.

Now, with hash_size of 9000 and 57 flow managers that looks like a lot: Every manager is allocated 157 rows, while the last one deals with 208 rows (~30% more).

However, seems like hash_size 9000 and 57 flow manager isn't a practical setting one would run.

With hash_size 65536 (default) and 7 flow managers (still quite high?), it's 9362 rows for flow-managers 1-6, and 9364 for manager 7, which is only 2 rows more and probably negligible.

Reply:
That logic/idea is already in place already in the current code. It's just more explicit now.

Reply:
@victorjulien / @inashivb - is there anything more you'd need input on for this PR?